src/content/content_location.rs

```diff
@@ -0,0 +1,118 @@
+use crate::headers::{HeaderName, HeaderValue, Headers, CONTENT_LOCATION};
+use crate::{Status, Url};
+
+use std::convert::TryInto;
+
+/// Indicates an alternate location for the returned data.
+///
+/// # Specifications
+///
+/// - [RFC 7231, section 3.1.4.2: Content-Location](https://tools.ietf.org/html/rfc7231#section-3.1.4.2)
+///
+/// # Examples
+///
+/// ```
+/// # fn main() -> http_types::Result<()> {
+/// #
+/// use http_types::{Response,Url};
+/// use http_types::content::{ContentLocation};
+///
+/// let content_location = ContentLocation::new(Url::parse("https://example.net/")?);
+///
+/// let mut res = Response::new(200);
+/// content_location.apply(&mut res);
+///
+/// let content_location = ContentLocation::from_headers(Url::parse("https://example.net/").unwrap(),res)?.unwrap();
+/// assert_eq!(content_location.location(), "https://example.net/");
+/// #
+/// # Ok(()) }
+/// ```
+#[derive(Debug)]
+pub struct ContentLocation {
+    url: Url,
+}
+
+impl ContentLocation {
+    /// Create a new instance of `Content-Location` header.
+    pub fn new(url: Url) -> Self {
+        Self { url }
+    }
+
+    /// Create a new instance from headers.
+    pub fn from_headers<U>(base_url: U, headers: impl AsRef<Headers>) -> crate::Result<Option<Self>>
```

Comment:
Is this correct? (Mostly hoping Yosh with double check this.)

Reply:
I think it is!